File: OIS/ois_progressiveart/progressiveart.c
// NOTE: it is recommended to use this even if you don't understand the following code.

#include <assert.h>
#include <stdio.h>
#include <stdlib.h>

#define MAXN 1000
#define MAXS 1000

int K, L, M, N, i;
char A[MAXN][MAXS];

int main() {
    // uncomment the two following lines if you want to read/write from files
    // freopen("input.txt", "r", stdin);
    // freopen("output.txt", "w", stdout);

    assert(4 == scanf("%d%d%d%d", &N, &M, &L, &K));

    // INSERT YOUR CODE HERE

    if (1) {
        printf("YES\n");
        for (i = 0; i < N; ++i)
            printf("%s\n", A[i]);
    } else {
        printf("NO\n");
    }

    return 0;
}

File: OIS/ois_progressiveart/progressiveart.pas
{ NOTE: it is recommended to use this even if you don't understand the following code }

const
    MAXN = 1000;

var
    K, L, M, N, i : LongInt;
    A             : Array[0..MAXN-1] of AnsiString;

begin
{
    uncomment the two following lines if you want to read/write from files
    assign(input,  'input.txt');  reset(input);
    assign(output, 'output.txt'); rewrite(output);
}

    ReadLn(N, M, L, K);

    for i := 0 to MAXN-1 do A[i] := '';

    { INSERT YOUR CODE HERE }

    WriteLn('YES');
    for i:=0 to N-1 do
        WriteLn(A[i]);

end.
